### PR TITLE
Replaced unnecessary map and getOrElse with exists

### PR DESCRIPTION
**Pull request by himani1**

### CODE REVIEW

File: core/src/main/scala/kafka/security/auth/SimpleAclAuthorizer.scala [hunk 1]
```diff
@@ -166,10 +166,10 @@ class SimpleAclAuthorizer extends Authorizer with Logging {
         && (acl.principal == principal || acl.principal == Acl.WildCardPrincipal)
         && (operations == acl.operation || acl.operation == All)
         && (acl.host == host || acl.host == Acl.WildCardHost)
-    ).map { acl: Acl =>
+    ).exists { acl: Acl =>
```
Comment (hachikuji) [inline core/src/main/scala/kafka/security/auth/SimpleAclAuthorizer.scala]: Kind of annoying that the `find` uses parenthesis. Maybe fix that too? Also, is the type necessary?

Comment (himani1) [reply core/src/main/scala/kafka/security/auth/SimpleAclAuthorizer.scala]: @hachikuji I have fixed the parenthesis used with find and removed type.

File: core/src/main/scala/kafka/security/auth/SimpleAclAuthorizer.scala [hunk 2]
```diff
@@ -160,17 +160,16 @@ class SimpleAclAuthorizer extends Authorizer with Logging {
     } else false
   }
 
-  private def aclMatch(session: Session, operations: Operation, resource: Resource, principal: KafkaPrincipal, host: String, permissionType: PermissionType, acls: Set[Acl]): Boolean = {
+  private def aclMatch(session: Session, operations: Operation, resource: Resource, principal: KafkaPrincipal, host: String, permissionType: PermissionType, acls: Set[Acl]): Boolean =
     acls.find ( acl =>
```
Comment (hachikuji) [inline core/src/main/scala/kafka/security/auth/SimpleAclAuthorizer.scala]: Did you remove the parenthesis in the wrong place? This is the one I was referring to. I think we should probably keep the braces around the method itself (I usually only remove them if it's a one-liner, but that might be my personal preference).

Comment (himani1) [reply core/src/main/scala/kafka/security/auth/SimpleAclAuthorizer.scala]: @hachikuji I am not able to understand. find method takes a function, how can I pass function without parenthesis to it ?
`def find(p: A => Boolean): Option[A]`
Can you please help me out?

Comment (hachikuji) [reply core/src/main/scala/kafka/security/auth/SimpleAclAuthorizer.scala]: Sorry, I was trying to suggest replacing the parenthesis with braces for consistency (somewhat annoyingly, you have to move the `&&` to the previous lines to make it work). It's not a big deal, so if it doesn't actually seem better, feel free to ignore.

Comment (himani1) [reply core/src/main/scala/kafka/security/auth/SimpleAclAuthorizer.scala]: @hachikuji Ok. I get it now. I have updated with the changes you suggested. Kindly please have a look and let me know if there's some mistake.

**Comment by hachikuji** [comment]
LGTM. Thanks for the cleanup!

**Pull request closed without merging**